README.rst

```diff
@@ -53,6 +53,36 @@ Now write a ``jenkins.yml`` file and open a PR::
 Many instructions are available. Just ask the bot by commenting ``jenkins:
 help`` in an open PR!
 
+The bot's story
+===============
+
+The bot executes the extensions listed in setup.py's entry point
+'jenkins_epo.bot.extensions'. Each entry point defines a stage number which is
+used by the bot to decide in which order to execute extensions.
+
+The ``bot.run()`` asyncio coroutine invoked by the ``jenkins-epo bot`` command
+creates a payload in ``self.current`` that will be passed in each extension's
+method call. Note that the ``LOOP=1`` environment variable will make the
+``jenkins-epo bot`` command to loop forever over calls of ``bot.run()``.
+
+First step of ``bot.run()`` is executing the ``ext.begin()`` method of each
+extension. This method should encapsulate any of the extension's variable
+initialization and eventually kick the payload out ouf ``bot.run()`` by raising
+the ``SkipHead`` exception. In this case, ``bot.run()`` just returns None and
+no other extension method is run at all with this payload until the bot's next
+loop.
+
+Then, ``bot.run()`` executes the ``ext.process_instruction(instruction)``
+extension method for each instruction parsed from the github comments by the
+bot. You'll note from examples such as the OPM extension that the bot gives
+instructions to itself in HTML comments when posting user feedback. This
+demonstrates how to maintain a state for an extension on a pull request.
+
+Finnaly, the ``ext.run()`` extension coroutine is called, again with the
+opportunity to raise ``SkipHead`` and directly go to the end of ``bot.run()``
```

Comment:
Je ferai un seul paragraphe pour `SkipHead`: *Extension can abort the pipeline either in `begin()` or `run()` by raising `SkipHead` ad-hoc exception.* etc.